rq/job.py

```diff
@@ -1054,30 +1355,38 @@ def register_dependency(self, pipeline: Optional['Pipeline'] = None):
             connection.sadd(self.dependencies_key, dependency_id)
 
     @property
-    def dependency_ids(self):
+    def dependency_ids(self) -> List[bytes]:
         dependencies = self.connection.smembers(self.dependencies_key)
         return [Job.key_for(_id.decode())
                 for _id in dependencies]
 
-    def dependencies_are_met(self, parent_job: Optional['Job'] = None,
-                             pipeline: Optional['Pipeline'] = None, exclude_job_id: str = None):
-        """Returns a boolean indicating if all of this job's dependencies are _FINISHED_
+    def dependencies_are_met(self, parent_job: Optional['Job'] = None, pipeline: Optional['Pipeline'] = None,
+                             exclude_job_id: Optional[str] = None) -> bool:
+        """Returns a boolean indicating if all of this job's dependencies are `FINISHED`
 
         If a pipeline is passed, all dependencies are WATCHed.
 
         `parent_job` allows us to directly pass parent_job for the status check.
         This is useful when enqueueing the dependents of a _successful_ job -- that status of
         `FINISHED` may not be yet set in redis, but said job is indeed _done_ and this
         method is _called_ in the _stack_ of its dependents are being enqueued.
+
+        Args:
+            parent_job (Optional[Job], optional): The parent Job. Defaults to None.
+            pipeline (Optional[Pipeline], optional): The Redis' pipeline. Defaults to None.
+            exclude_job_id (Optional[str], optional): Whether to exclude the job id.. Defaults to None.
+
+        Returns:
+            are_met (bool): Whether the dependencies were met.
         """
         connection = pipeline if pipeline is not None else self.connection
 
         if pipeline is not None:
             connection.watch(*[self.key_for(dependency_id)
                                for dependency_id in self._dependency_ids])
 
-        dependencies_ids = {_id.decode()
-                            for _id in connection.smembers(self.dependencies_key)}
+        saved_job_dependents = connection.smembers(self.dependencies_key)
+        dependencies_ids = {_id.decode() for _id in saved_job_dependents}
```

Comment:
I think it's better if we stick with set comprehension here, it avoids the variable assignment `saved_job_dependents`

Reply:
I personally feel separating this is a great win for style, easy of understanding, readability and maintenance, but I can roll this back.